src/test/java/org/jabref/logic/remote/RemoteCommunicationTest.java

```diff
@@ -69,7 +69,17 @@ void commandLineArgumentTwoPassedToServer() {
     void commandLineArgumentMultiLinePassedToServer() {
         final String[] message = new String[]{"my message\n second line", "second \r and third"};
 
-        client.sendCommandLineArguments(message);
+        client.sendCommandLineArguments(message.clone());
+
+        verify(server).handleCommandLineArguments(message);
+    }
+
+    @Test
+    void commandLineArgumentEncodingAndDecoding() {
+        final String[] message = new String[]{"D:\\T EST\\测试te st.bib"};
+
+        // will be encoded as "D%3A%5CT+EST%5C%E6%B5%8B%E8%AF%95te+st.bib"
+        client.sendCommandLineArguments(message.clone());
 
         verify(server).handleCommandLineArguments(message);
     }
```

Comment:
I am wondering whether this test case is appropriate. Not so sure about how to write a test about socket communication.

Reply:
Yeah, it's hard to test this. I think, your added test is fine.